Tests/SPMBuildCoreTests/PluginInvocationTests.swift

```diff
@@ -89,17 +89,18 @@ class PluginInvocationTests: XCTestCase {
 
         // A fake PluginScriptRunner that just checks the input conditions and returns canned output.
         struct MockPluginScriptRunner: PluginScriptRunner {
-            
+
             var hostTriple: Triple {
                 return UserToolchain.default.triple
             }
             
-            func compilePluginScript(sources: Sources, toolsVersion: ToolsVersion, observabilityScope: ObservabilityScope) throws -> PluginCompilationResult {
+            func compilePluginScript(sourceFiles: [AbsolutePath], pluginName: String, toolsVersion: ToolsVersion, observabilityScope: ObservabilityScope) throws -> PluginCompilationResult {
```

Comment:
nit: these kind of functions that involve I/O seems like they should be async

Reply:
I think the client calls that used this have been removed, so I can probably remove these.

Reply:
Removed it.

Reply:
I pushed the `tsc_await` to all the call sites.